### PR TITLE
[20130] Implement TypeLookupManager async_get_type

### PR DESCRIPTION
**Pull request by adriancampo**
## Contributor Checklist
- [X] Commit messages follow the project guidelines. 
- [X] The code follows the style guidelines of this project. 
- N/A Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- N/A Any new/modified methods have been properly documented using Doxygen. 
- N/A Changes are ABI compatible. 
- N/A Changes are API compatible. 
- N/A New feature has been added to the `versions.md` file (if applicable).
- N/A New feature has been documented/Current behavior is correctly described in the documentation. 
    
- N/A Applicable backports have been included in the description.

## Reviewer Checklist
- [ ] The PR has a milestone assigned.
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupReplyListener.hpp [hunk 1]
```diff
@@ -48,10 +48,10 @@ class TypeLookupReplyListener : public fastrtps::rtps::ReaderListener
 
     /**
      * @brief Constructor
-     * @param pwlp Pointer to the writer liveliness protocol
+     * @param pwlp Pointer to the TypeLookupManager
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/builtin/type_lookup_service/TypeLookupReplyListener.hpp:51]:
```suggestion
     * @param manager Pointer to the TypeLookupManager
```

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupRequestListener.hpp [hunk 2]
```diff
@@ -48,10 +48,10 @@ class TypeLookupRequestListener : public fastrtps::rtps::ReaderListener
 
     /**
      * @brief Constructor
-     * @param pwlp Pointer to the writer liveliness protocol
+     * @param pwlp Pointer to the TypeLookupManager
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/builtin/type_lookup_service/TypeLookupRequestListener.hpp:51]:
```suggestion
     * @param manager Pointer to the TypeLookupManager
```

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp [hunk 3]
```diff
@@ -46,6 +46,8 @@ struct hash<eprosima::fastdds::dds::xtypes::TypeIdentifier>
 
 };
 
+#ifndef TYPE_IDENTIFIER_WITH_SIZE_HASH_SPECIALIZATION
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp:49]: Define this hash in a common header.

File: src/cpp/rtps/participant/RTPSParticipantImpl.h [hunk 4]
```diff
@@ -49,6 +49,7 @@
 #include <fastrtps/utils/shared_mutex.hpp>
 
 #include "../flowcontrol/FlowControllerFactory.hpp"
+#include <fastdds/builtin/type_lookup_service/TypeLookupManager.hpp>
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/rtps/participant/RTPSParticipantImpl.h:52]: It is not necessary as it has been forward declared. It can be removed.

File: src/cpp/rtps/participant/RTPSParticipantImpl.cpp [hunk 5]
```diff
@@ -62,6 +62,7 @@
 #include <rtps/persistence/PersistenceService.h>
 #include <statistics/rtps/GuidUtils.hpp>
 
+
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/rtps/participant/RTPSParticipantImpl.cpp:65]: Please, revert this change as it is not including nothing new to the PR.

File: src/cpp/rtps/builtin/discovery/endpoint/EDPSimpleListeners.cpp
Comment (JLBuenoLopez-eProsima) [inline src/cpp/rtps/builtin/discovery/endpoint/EDPSimpleListeners.cpp:1]: Have you taken a look into the EDPServerListeners files? These are applied when Discovery Server is used. In that case this feature should also work.

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp [hunk 6]
```diff
@@ -49,8 +50,44 @@ namespace fastdds {
...
+    fastrtps::rtps::GUID_t rtps_guid;
+    std::memcpy(rtps_guid.guidPrefix.value, guid.guidPrefix().data(), 12);
+    for (size_t i = 0; i < 3; i++)
+    {
+        rtps_guid.entityId.value[i + 1] = guid.entityId().entityKey()[i];
+    }
+    rtps_guid.entityId.value[0] = guid.entityId().entityKind();
+
+    return rtps_guid;
+}
```
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp]: Changed in [ee520e4](https://github.com/eProsima/Fast-DDS/pull/4177/commits/ee520e48f15c205983ef93344d678cca12d35d0b)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp [hunk 7]
```diff
@@ -235,10 +272,10 @@ void TypeLookupManager::remove_remote_endpoints(
     }
 }
 
-fastrtps::rtps::SampleIdentity TypeLookupManager::get_type_dependencies(
+SampleIdentity TypeLookupManager::get_type_dependencies(
         const xtypes::TypeIdentifierSeq& id_seq) const
 {
-    fastrtps::rtps::SampleIdentity id = INVALID_SAMPLE_IDENTITY;
+    SampleIdentity id = builtin::INVALID_SAMPLE_IDENTITY;
```
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp]: Changed in [ee520e4](https://github.com/eProsima/Fast-DDS/pull/4177/commits/ee520e48f15c205983ef93344d678cca12d35d0b)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp [hunk 8]
```diff
@@ -267,18 +304,156 @@ fastrtps::rtps::SampleIdentity TypeLookupManager::get_types(
...
 
 ReturnCode_t TypeLookupManager::async_get_type(
-        xtypes::TypeInformation,
-        GuidPrefix_t,
-        AsyncGetTypeCallback&)
+        xtypes::TypeInformation type_inf,
+        fastrtps::rtps::GuidPrefix_t type_server,
+        AsyncGetTypeCallback& callback)
+{
+    ReturnCode_t solve_ret = solve_type(type_inf.complete().typeid_with_size());
```
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp]: Changed in [ee520e4](https://github.com/eProsima/Fast-DDS/pull/4177/commits/ee520e48f15c205983ef93344d678cca12d35d0b)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp [hunk 9]
```diff
@@ -161,47 +175,119 @@ class TypeLookupManager
...
+    ReturnCode_t solve_type(
+            xtypes::TypeIdentfierWithSize type_id);
+
+    /**
+     *  Notifies callbacks for a given TypeIdentfierWithSize
+     * @param type_id[in] TypeIdentfierWithSize of the callbacks to notify.
+     */
+    void notify_callbacks(
+            xtypes::TypeIdentfierWithSize type_id);
+    /**
```
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp:203]: Added in [ee520e4](https://github.com/eProsima/Fast-DDS/pull/4177/commits/ee520e48f15c205983ef93344d678cca12d35d0b)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp [hunk 10]
```diff
@@ -161,47 +175,119 @@ class TypeLookupManager
...
+     * Adds a callback to the async_get_type_callbacks_ entry of the TypeIdentfierWithSize, or creates a new one if
+     * TypeIdentfierWithSize was not in the map before
```
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp:205]: Fixed in [ee520e4](https://github.com/eProsima/Fast-DDS/pull/4177/commits/ee520e48f15c205983ef93344d678cca12d35d0b)

File: src/cpp/rtps/participant/RTPSParticipantImpl.cpp (hunk 5 above)
Comment (adriancampo) [reply src/cpp/rtps/participant/RTPSParticipantImpl.cpp]: removed in [ee520e4](https://github.com/eProsima/Fast-DDS/pull/4177/commits/ee520e48f15c205983ef93344d678cca12d35d0b)

File: src/cpp/rtps/participant/RTPSParticipantImpl.h (hunk 4 above)
Comment (adriancampo) [reply src/cpp/rtps/participant/RTPSParticipantImpl.h]: Removed in [ee520e4](https://github.com/eProsima/Fast-DDS/pull/4177/commits/ee520e48f15c205983ef93344d678cca12d35d0b)

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp (hunk 3 above)
Comment (adriancampo) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp]: Created header in [ee520e4](https://github.com/eProsima/Fast-DDS/pull/4177/commits/ee520e48f15c205983ef93344d678cca12d35d0b)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp [hunk 11]
```diff
@@ -20,21 +20,84 @@
...
 #define _FASTDDS_TYPELOOKUP_SERVICE_MANAGER_HPP
 
-#include <vector>
 #include <mutex>
 #include <unordered_map>
+#include <vector>
 
 #include <fastdds/rtps/builtin/data/ReaderProxyData.h>
 #include <fastdds/rtps/builtin/data/WriterProxyData.h>
+#include <fastrtps/types/TypeObject.h>
```
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp]: Removed in [ee520e4](https://github.com/eProsima/Fast-DDS/pull/4177/commits/ee520e48f15c205983ef93344d678cca12d35d0b)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp [hunk 12]
```diff
@@ -20,21 +20,84 @@
...
+struct hash<eprosima::fastdds::dds::xtypes::TypeIdentfierWithSize>
+{
+    std::size_t operator ()(
+            const eprosima::fastdds::dds::xtypes::TypeIdentfierWithSize& k) const
+    {
+        return static_cast<size_t>(k.typeobject_serialized_size());
+    }
+
+};
+#endif // TYPE_IDENTIFIER_WITH_SIZE_HASH_SPECIALIZATION
```
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp]: Created header in [ee520e4](https://github.com/eProsima/Fast-DDS/pull/4177/commits/ee520e48f15c205983ef93344d678cca12d35d0b)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp [hunk 13]
```diff
@@ -161,47 +175,119 @@ class TypeLookupManager
...
      */
-    fastrtps::rtps::SampleIdentity get_types(
+    SampleIdentity get_types(
             const xtypes::TypeIdentifierSeq& id_seq) const;
 
     /**
      * Use builtin TypeLookup service to solve the type and dependencies of a given TypeInformation.
      * It receives a callback that will be used to notify when the negotiation is complete.
-     * @param typeinformation[in] TypeInformation that requires solving.
+     * @param type_inf[in] TypeInformation that requires solving.
```
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp]: Changed in [ee520e4](https://github.com/eProsima/Fast-DDS/pull/4177/commits/ee520e48f15c205983ef93344d678cca12d35d0b)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp [hunk 14]
```diff
@@ -267,18 +304,156 @@ fastrtps::rtps::SampleIdentity TypeLookupManager::get_types(
...
+        xtypes::TypeIdentfierWithSize type_id)
+{
+    xtypes::TypeIdentifierSeq uknown_type;
+
+    // Check if the type is known
+    if (fastrtps::rtps::RTPSDomainImpl::get_instance()->type_object_registry_observer().
+                    is_type_identifier_known(type_id.type_id()))
+    {
+        return RETCODE_OK;
+    }
```
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp:341]: See [a8ecf0d](https://github.com/eProsima/Fast-DDS/pull/4177/commits/a8ecf0de2a5b4909201d89840a8e0a7b23147de2)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp [hunk 15]
```diff
@@ -267,18 +304,156 @@ fastrtps::rtps::SampleIdentity TypeLookupManager::get_types(
...
+    else
+    {
+        uknown_type.push_back(type_id.type_id());
+        SampleIdentity get_type_dependencies_request = get_type_dependencies(uknown_type);
+        if (builtin::INVALID_SAMPLE_IDENTITY != get_type_dependencies_request)
+        {
+            // Store the sent requests and associated TypeIdentfierWithSize
+            add_async_get_type_request(get_type_dependencies_request, type_id);
```
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp]: When the requests are received we need to have find the TypeIdentfierWithSize that originated them

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp [hunk 16]
```diff
@@ -267,18 +304,156 @@ fastrtps::rtps::SampleIdentity TypeLookupManager::get_types(
...
+        async_get_type_requests_.erase(request);
+        return true;
+    }
+    catch (const std::exception& e)
+    {
+        EPROSIMA_LOG_ERROR(TYPELOOKUP_SERVICE,
+                "Error in TypeLookupManager::remove_async_get_types_request: " << e.what());
+        return false;
+    }
 }
```
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp]: Removed in [ee520e4](https://github.com/eProsima/Fast-DDS/pull/4177/commits/ee520e48f15c205983ef93344d678cca12d35d0b)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp [hunk 17]
```diff
@@ -267,18 +304,156 @@ fastrtps::rtps::SampleIdentity TypeLookupManager::get_types(
...
+}
+
+ReturnCode_t TypeLookupManager::solve_type(
+        xtypes::TypeIdentfierWithSize type_id)
+{
+    xtypes::TypeIdentifierSeq uknown_type;
+
+    // Check if the type is known
+    if (fastrtps::rtps::RTPSDomainImpl::get_instance()->type_object_registry_observer().
+                    is_type_identifier_known(type_id.type_id()))
```
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp]: Fixed in [ee520e4](https://github.com/eProsima/Fast-DDS/pull/4177/commits/ee520e48f15c205983ef93344d678cca12d35d0b)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp [hunk 18]
```diff
@@ -267,18 +304,156 @@ fastrtps::rtps::SampleIdentity TypeLookupManager::get_types(
...
+    if (callbacks_it != async_get_type_callbacks_.end())
+    {
+        for (AsyncGetTypeCallback& callback : callbacks_it->second)
+        {
+            callback();
+        }
+        // Erase the solved TypeInformation
+        remove_async_get_type_callback(type_id);
+    }
+}
```
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp]: Removed in [a8ecf0d](https://github.com/eProsima/Fast-DDS/pull/4177/commits/a8ecf0de2a5b4909201d89840a8e0a7b23147de2)

File: src/cpp/rtps/builtin/discovery/endpoint/EDPSimpleListeners.cpp
Comment (adriancampo) [reply src/cpp/rtps/builtin/discovery/endpoint/EDPSimpleListeners.cpp:1]: They use the functions from EDPSimpleListeners

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp [hunk 19]
```diff
@@ -267,18 +304,156 @@ fastrtps::rtps::SampleIdentity TypeLookupManager::get_types(
...
+        async_get_type_requests_.emplace(request, type_id);
+        return true;
+    }
+    catch (const std::exception& e)
+    {
+        EPROSIMA_LOG_ERROR(TYPELOOKUP_SERVICE,
+                "Error in TypeLookupManager::add_async_get_type_request: " << e.what());
+        return false;
+    }
+}
```
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp:415]: When the requests are received we need to have find the TypeIdentfierWithSize that originated them

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp [hunk 20]
```diff
@@ -161,47 +175,119 @@ class TypeLookupManager
...
+    /**
+     * Adds a callback to the async_get_type_callbacks_ entry of the TypeIdentfierWithSize, or creates a new one if
+     * TypeIdentfierWithSize was not in the map before
+     * @param request[in] SampleIdentity of the request
+     * @param type_id[in] TypeIdentfierWithSize that originated the request.
+     * @return true if added. false otherwise
+     */
+    bool add_async_get_type_request(
+            SampleIdentity request,
+            xtypes::TypeIdentfierWithSize type_id );
```
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp:216]: When the requests are received we need to have find the TypeIdentfierWithSize that originated them

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp [hunk 21]
```diff
@@ -265,6 +357,15 @@ class TypeLookupManager
...
     mutable TypeLookup_ReplyPubSubType reply_type_;
+
+    //!Mutex to protect access to async_get_type_callbacks_ and async_get_type_requests_
+    std::mutex async_get_types_mutex_;
+
+    //!Collection of all the callbacks related to a TypeIdentfierWithSize, hashed by its TypeIdentfierWithSize.
+    std::unordered_map<xtypes::TypeIdentfierWithSize, std::vector<AsyncGetTypeCallback>> async_get_type_callbacks_;
+
+    //!Collection SampleIdentity and the TypeIdentfierWithSize it originated from, hashed by its SampleIdentity.
+    std::unordered_map<SampleIdentity, xtypes::TypeIdentfierWithSize> async_get_type_requests_;
```
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp:337]: When the requests are received we need to have find the TypeIdentfierWithSize that originated them

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp [hunk 22]
```diff
@@ -522,18 +697,23 @@ bool TypeLookupManager::recv_request(
...
-    msg.pos += 2; // Skip encapsulation options.
+    msg.pos += 2;     // Skip encapsulation options.
 
     SerializedPayload_t payload;
     payload.max_size = change.serializedPayload.max_size - 4;
     payload.length = change.serializedPayload.length - 4;
     payload.data = change.serializedPayload.data + 4;
     bool result = request_type_.deserialize(&payload, &request);
     payload.data = nullptr;
+    if (result && dds_2_rtps(request.header().requestId().writer_guid()) == builtin_request_writer_->getGuid())
```
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp]: Fixed in [e617563](https://github.com/eProsima/Fast-DDS/pull/4177/commits/e617563d450062d7d570948f1953f992049b560b)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp [hunk 23]
```diff
@@ -311,79 +311,70 @@ SampleIdentity TypeLookupManager::get_types(
...
     if (fastrtps::rtps::RTPSDomainImpl::get_instance()->type_object_registry_observer().
-                    is_type_identifier_known(type_id.type_id()))
+                    is_type_identifier_known(type_identifier_with_size))
     {
         return RETCODE_OK;
     }
-    else
+
+    // Check if the TypeInformation was received before
+    if (async_get_type_callbacks_.find(type_identifier_with_size) != async_get_type_callbacks_.end())
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp:350]: In this case you should register this new callback in the async_get_type_callbacks_ collection, within the same TypeIdentfierWithSize key, adding a new element to the AsyncGetTypeCallback vector. Consequently, to avoid multiple searchs, the iterator can be saved in a variable, check if it is the end iterator or not, and if it is not, accesing the iterator element and adding the information to the corresponding vector.

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp [hunk 24]
```diff
@@ -235,50 +274,167 @@ void TypeLookupManager::remove_remote_endpoints(
...
+    std::unique_lock<std::mutex> lock(async_get_types_mutex_);
+    try
+    {
+        // Check if TypeIdentfierWithSize already exists in the map
+        auto it = async_get_type_callbacks_.find(type_identifier_with_size);
+        if (it != async_get_type_callbacks_.end())
+        {
+            // TypeIdentfierWithSize exists, add the callback
+            it->second.push_back(callback);
+        }
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp:388]: This search has already been performed in `check_type_identifier_received`. I suggest  adding the callback in that other method so the same search is avoided and done only once.

File: src/cpp/fastdds/xtypes/type_representation/TypeIdentifierWithSizeHashSpecialization.h [hunk 25]
```diff
@@ -0,0 +1,43 @@
+// Copyright 2016 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/xtypes/type_representation/TypeIdentifierWithSizeHashSpecialization.h:1]:
```suggestion
// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp [hunk 26]
```diff
@@ -29,6 +29,7 @@
 #include <fastdds/dds/xtypes/type_representation/ITypeObjectRegistry.hpp>
 #include <fastdds/dds/xtypes/type_representation/TypeObject.hpp>
 #include <fastdds/dds/xtypes/type_representation/TypeObjectUtils.hpp>
+#include <fastdds/xtypes/type_representation/TypeIdentifierWithSizeHashSpecialization.h>
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp:32]: Private headers are included in a different block below the library public headers.

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp [hunk 27]
```diff
@@ -161,47 +175,119 @@ class TypeLookupManager
...
+     * @param request[in] SampleIdentity to be removed.
+     * @return true if removed. false otherwise
+     */
+    bool remove_async_get_types_request(
+            SampleIdentity request);
 
-    //! Aux method to send requests
+    /**
+     * Complete requests common fields, create CacheChange, serialize request and add change to writer history.
+     * @param request[in] TypeLookup_Request to be send.
```
[the remark this answers is not in the document]
Comment (JLBuenoLopez-eProsima) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp]: There are still some other occurrences of this typo. Please, fix

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp [hunk 28]
```diff
@@ -235,50 +274,167 @@ void TypeLookupManager::remove_remote_endpoints(
...
+    else
+    {
+        // Failed to send request, return error
+        return RETCODE_ERROR;
+    }
+}
+
+bool TypeLookupManager::add_async_get_type_callback(
+        const xtypes::TypeIdentfierWithSize& type_identifier_with_size,
+        const fastrtps::rtps::GuidPrefix_t& type_server,
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp:375]: If this parameter is not used, remove it.

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp [hunk 29]
```diff
@@ -214,12 +276,26 @@ class TypeLookupManager
         return participant_;
     }
 
-    //! Get out instanceName as defined in 7.6.3.3.4 the XTypes 1.3 document
-    std::string get_instanceName() const;
+    /**
+     * Create instance name as defined in 7.6.3.3.4 the XTypes 1.3 document
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp:280]:
```suggestion
     * Create instance name as defined in section 7.6.3.3.4 XTypes 1.3 specification

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp [hunk 30]
```diff
@@ -281,6 +293,9 @@ class TypeLookupManager
     //!Pointer to the local RTPSParticipant.
     fastrtps::rtps::RTPSParticipantImpl* participant_ = nullptr;
 
+    //!Own instance name
+    std::string own_instance_name_;
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp:297]: I suggest renaming to `local_instance_name`.

Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp]: Renamed in [cfa3f78](https://github.com/eProsima/Fast-DDS/pull/4177/commits/cfa3f781c09d678528b673acc9feef170c546bc0)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp (hunk 29 above)
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp]: Updated in [cfa3f78](https://github.com/eProsima/Fast-DDS/pull/4177/commits/cfa3f781c09d678528b673acc9feef170c546bc0)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp (hunk 28 above)
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp]: Removed in [cfa3f78](https://github.com/eProsima/Fast-DDS/pull/4177/commits/cfa3f781c09d678528b673acc9feef170c546bc0)

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp (hunk 26 above)
Comment (adriancampo) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.hpp]: Moved in [cfa3f78](https://github.com/eProsima/Fast-DDS/pull/4177/commits/cfa3f781c09d678528b673acc9feef170c546bc0)

File: src/cpp/fastdds/xtypes/type_representation/TypeIdentifierWithSizeHashSpecialization.h (hunk 25 above)
Comment (adriancampo) [reply src/cpp/fastdds/xtypes/type_representation/TypeIdentifierWithSizeHashSpecialization.h]: Updated in [cfa3f78](https://github.com/eProsima/Fast-DDS/pull/4177/commits/cfa3f781c09d678528b673acc9feef170c546bc0)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp (hunk 24 above)
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp]: Removed method in [cfa3f78](https://github.com/eProsima/Fast-DDS/pull/4177/commits/cfa3f781c09d678528b673acc9feef170c546bc0)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp (hunk 23 above)
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp]: Changed in [cfa3f78](https://github.com/eProsima/Fast-DDS/pull/4177/commits/cfa3f781c09d678528b673acc9feef170c546bc0)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp [hunk 31]
```diff
@@ -51,7 +51,7 @@ namespace dds {
 namespace builtin {
 
 
-inline SequenceNumber_t rtps_2_dds(
+inline SequenceNumber_t sequence_number_rtps_2_dds(
```
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp:54]: Looks super scary, i like it :P
Updated in [cfa3f78](https://github.com/eProsima/Fast-DDS/pull/4177/commits/cfa3f781c09d678528b673acc9feef170c546bc0)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp (hunk 27 above)
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.hpp]: Fised in [cfa3f78](https://github.com/eProsima/Fast-DDS/pull/4177/commits/cfa3f781c09d678528b673acc9feef170c546bc0)

File: src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp [hunk 32]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2019 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/builtin/type_lookup_service/TypeLookupManager.cpp]: Changed in [6283119](https://github.com/eProsima/Fast-DDS/pull/4177/commits/62831190e66bd594d0da9f0996c7ab07441dceef)

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM with green CI (build)

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

**Comment by adriancampo** [comment]
@richiprosima please test this

**Comment by adriancampo** [comment]
@richiprosima please test this

**Comment by adriancampo** [comment]
@richiprosima please test mac

**Comment by adriancampo** [comment]
@richiprosima please test this

**Comment by adriancampo** [comment]
@richiprosima please test mac

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
Changes LGTM with green CI (build)

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

**Comment by adriancampo** [comment]
@richiprosima please test mac

**Comment by adriancampo** [comment]
@richiprosima please test mac
